tests/Silverback.Integration.Tests.E2E/Kafka/CallbacksTests.cs

```diff
@@ -171,6 +347,26 @@ private class AnotherSimpleCallbackHandler : SimpleCallbackHandler
         {
         }
 
+        [SuppressMessage("", "CA1812", Justification = Justifications.CalledBySilverback)]
+        [SuppressMessage(
+            "ReSharper",
+            "ClassNeverInstantiated.Local",
+            Justification = Justifications.CalledBySilverback)]
+        private class KafkaEndOfTopicPartitionReachedCallback : IKafkaEndOfTopicPartitionReachedCallback
+        {
+            private readonly object _lockObject = new();
+
+            public int EndOfPartitionReachedCount { get; private set; }
+
+            public void OnEndOfTopicPartitionReached(TopicPartition topicPartition, KafkaConsumer consumer)
+            {
+                lock (_lockObject)
```

Comment:
Don't use a lock to increment an integer, switch to a property with an explicit backing field and use `Interlocked.Increment` which is more expressive.